qadence/ml_tools/train_grad.py

```diff
@@ -177,16 +198,28 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
                 if iteration % config.write_every == 0:
                     write_tensorboard(writer, loss, metrics, iteration)
 
+                if perform_val:
+                    if iteration % config.val_every == 0:
+                        xs = next(dl_iter_val)
+                        xs_to_device = data_to_device(xs, device=device, dtype=data_dtype)
+                        val_loss, _ = loss_fn(model, xs_to_device)
+                        if config.validation_criterion(val_loss, best_val_loss, config.val_epsilon):
+                            best_val_loss = val_loss
+                            if config.folder and config.checkpoint_best_only:
+                                write_checkpoint(config.folder, model, optimizer, iteration="best")
+                            metrics["val_loss"] = val_loss
+                            write_tensorboard(writer, math.nan, metrics, iteration)
```

Comment:
why `math.nan`?

Reply:
`write_tensorboard(writer, None, metrics, iteration)` with `None` instead of NaN raised an error. This is the traceback.

```bash
    def make_np(x):
        """
        Convert an object into numpy array.
    
        Args:
          x: An instance of torch tensor or caffe blob name
    
        Returns:
            numpy.array: Numpy array
        """
        if isinstance(x, np.ndarray):
            return x
        if isinstance(x, str):  # Caffe2 will pass name of blob(s) to fetch
            return _prepare_caffe2(x)
        if np.isscalar(x):
            return np.array([x])
        if isinstance(x, torch.Tensor):
            return _prepare_pytorch(x)
>       raise NotImplementedError(
            f"Got {type(x)}, but numpy array, torch tensor, or caffe2 blob name are expected."
        )
E       NotImplementedError: Got <class 'NoneType'>, but numpy array, torch tensor, or caffe2 blob name are expected.

../../.local/share/hatch/env/virtual/qadence/G8biN769/qadence/lib/python3.11/site-packages/torch/utils/tensorboard/_convert_np.py:24: NotImplementedError
```

That's why I needed to change [this line in `printing.py`](https://github.com/pasqal-io/qadence/pull/435/files#diff-409c355dde25623d704f5e5d205ab9043700fff703bd4d28443f35d186d95dd1L15-R15). I have `tensorboard==2.16.2`.

Reply:
interesting, but thanks for fixing